dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
```

Comment:
I will create a separate PR for Signal class.

> The only question left would be what to name the Subscriber/Publisher classes because I think those names sound too general for something with such a specific purpose.

Agree. The key function of Publisher is `sendDestructionNotification()` so it could be like DestructionNoticable/DestructionObserver.